sympy/stats/crv_types.py

```diff
@@ -1379,8 +1379,24 @@ def pdf(self, x):
         k, theta = self.k, self.theta
         return x**(k - 1) * exp(-x/theta) / (gamma(k)*theta**k)
 
-    def sample(self):
-        return random.gammavariate(self.k, self.theta)
+    def _sample_random(self, size):
+        return [random.gammavariate(self.k, self.theta) for i in range(size)]
+
+    def _sample_numpy(self, numpy, size):
+        k, theta = float(self.k), float(self.theta)
+        return numpy.random.gamma(k, theta, size)
+
+    def _sample_scipy(self, scipy, size):
+        k, theta = float(self.k), float(self.theta)
+        from scipy.stats import gamma
+        return gamma.rvs(a=k, loc=0, scale=theta, size=size)
```

Comment:
Is it possible that `theta` and `size` to be symbolic?

Reply:
Well, if you are going to sample, I would say you need all variables to be numeric.

Reply:
Do you see a way to numerically sample distributions with symbolic parameters? I think it could be done in certain cases, but I would recommend discussing this on gitter first.

Reply:
Yes, I agree with @Upabjojr that the variable should be numeric, but for some distributions like `NormalDistribution`, `GammaDistribution` etc we have used `return random.<distr>()` which allows for symbolic parameters but only gives one sample per call.
I have kept the function for cases where the sampling library is not mentioned and such function exist. Please tell if you want it removed.

Reply:
So, there are chances that parameter can be symbolic. I think we can add a check which can raise an informative message before passing them to `gamma.rvs`. Though we expect from the user to take care of that but may be if this feature is used in a large code base then may be some conditions may allow symbolic parameters to get passed to `gamma.rvs`. 
PS - This is just a suggestion. Infact, `gamma.rvs` will raise error if `theta` or `size` are symbolic, however that will not be very much informative.
You can do something like,
```python
if theta.has(Symbol) or size.has(Symbol):
    raise ValueError("Sampling is currently not possible for symbolic parameters.")
```

Reply:
@czgdp1807 this is the work of the dispatcher which will look for symbolic parameters. If the parameters are symbolic or no sample library is mentioned then `_sample_random` will be called (`_sample_random` will return an array if `random` could sample the distribution, otherwise will throw an error).
Please let me know what you think of this approach.

Reply:
Oh I see, no issues then. In fact, your approach will be better in avoiding the repetition of the checking code. From my side, this conversation can be resolved.